### PR TITLE
Controllerify attack

### LINKED ISSUE #396: As a developer, I find all of the request handling logic for combat in a single controller
#### AC:
1. attack_mod.php is deleted
2. combat functionality still works

### PR DESCRIPTION
**Pull request by Beagle**
resolve #396

### CODE REVIEW

File: deploy/lib/control/AttackController.php [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+            }
+        }
+
+        $target_ending_health = $target_player->health();
+        $target_name = $target_player->name();
+        $target_id = $target_player->id();
+        return [
+            'template' => 'attack_mod.tpl',
+            'title'    => 'Battle Status',
+            'parts'    => get_defined_vars(),
```
Comment (tchalvak) [inline deploy/lib/control/AttackController.php]: implicit get_defined_vars is a problem that we def have to deal with down the road.

Comment (Beagle) [inline deploy/lib/control/AttackController.php]: Yup, I've already begun by eliminating redundant variables. I accomplished
this for the InventoryController in my refactor of that, I expect we'll
accomplish the same thing for combat soon enough.

**Comment by tchalvak** [comment]
Any way to write any sanity tests on it?

**Comment by tchalvak** [comment]
Tested, was able to evade punks, blaze, deflect, duel, single attack, die, kill, tie, obtain gold, lose gold, be wrathful.  Was not able to gain ki?

Bounty-obtaining is broken, have not yet ascertained whether it is a pre-existing condition.

"You have received the 0 gold bounty on sawako's head for your deeds!"

screenshot from 2016-03-18 01 18 40

Tested that on live, it's a pre-existing condition where you can't get bounty, or at least it reports badly as not receiving bounty.

Other than that:
red
green
refactor-all-the-things

**Pull request merged**